docs/development/release.md

```diff
@@ -3,148 +3,297 @@
 
 ## Prerequisite
 
-You need [KubeRay GitHub](https://github.com/ray-project/kuberay) write permissions to cut a release branch and create a release tag.
+You need write access to the [KubeRay repo](https://github.com/ray-project/kuberay) to cut a release branch and create a release tag.
 
-### Overview
+## Overview
 
-Each major release (e.g. `0.4`) is managed in its own GitHub branch.
-To release KubeRay, cut a release branch (e.g. `release-0.4`) from master and build commits on that branch
-until you reach a satisfactory final release commit.
+Each KubeRay minor release series (e.g., `v1.3.X`) is maintained on its own release branch. For example, the `v1.3.X` releases are cut from the `release-1.3` branch. Patch releases (e.g., `v1.3.1`) involve cherry-picking commits onto the corresponding release branch.
 
-Immediately after cutting the release branch, create a commit for a release candidate (e.g. `0.4.0-rc.0`),
-and build the associated artifacts (images and charts).
-If further changes need to be made to the release, pick changes from the master branch into the release branch.
-Make as many release candidates as necessary until a stable final release commit is reached.
-Then build final release artifacts, publish release notes, and announce the release.
+The high-level steps for a new minor or patch release are:
 
-### KubeRay release schedule
+1. For new minor releases, create a release branch (e.g., `release-1.4`) from `master` if it doesn't exist.
```

Comment:
This flow does not include the process for releasing RCs. The details regarding the version strings for images and Helm charts are important.

Reply:
The process is the same. I'll add a short note about it so the tag format is clear